cvat/apps/analytics_report/report/create.py

```diff
@@ -0,0 +1,319 @@
+            scheduler.enqueue_at(
+                next_job_time,
+                self._check_job_analytics,
+                cvat_task_id=cvat_task_id,
+                cvat_project_id=cvat_project_id,
+                job_id=queue_job_id,
+            )
+
+    def schedule_analytics_check_job(self, *, job=None, task=None, project=None, user_id):
+        rq_id = self._make_custom_analytics_check_job_id()
+
+        queue = self._get_queue()
+        queue.enqueue(
+            self._check_job_analytics,
+            cvat_job_id=job.id if job is not None else None,
+            cvat_task_id=task.id if task is not None else None,
+            cvat_project_id=project.id if project is not None else None,
+            job_id=rq_id,
+            meta={"user_id": user_id, "job_type": self._RQ_CUSTOM_ANALYTICS_CHECK_JOB_TYPE},
+            result_ttl=self._JOB_RESULT_TTL,
+            failure_ttl=self._JOB_RESULT_TTL,
+        )
+
+        return rq_id
+
+    def get_analytics_check_job(self, rq_id: str):
+        queue = self._get_queue()
+        rq_job = queue.fetch_job(rq_id)
+
+        if rq_job and not self.is_custom_analytics_check_job(rq_job):
+            rq_job = None
+
+        return rq_job
+
+    def is_custom_analytics_check_job(self, rq_job) -> bool:
+        return rq_job.meta.get("job_type") == self._RQ_CUSTOM_ANALYTICS_CHECK_JOB_TYPE
+
+    @classmethod
+    def _check_job_analytics(cls, *, cvat_job_id: int=None, cvat_task_id: int=None, cvat_project_id: int=None) -> int:
+        if cvat_job_id is not None:
+            db_job = Job.objects.select_related("analytics_report").get(pk=cvat_job_id)
+            return cls()._compute_report_for_job(db_job)
+        elif cvat_task_id is not None:
+            db_task = Task.objects.select_related("analytics_report").prefetch_related("segment_set__job_set").get(pk=cvat_task_id)
+            return cls()._compute_report_for_task(db_task)
+        elif cvat_project_id is not None:
+            db_project = Project.objects.select_related("analytics_report").prefetch_related("tasks__segment_set__job_set").get(pk=cvat_project_id)
+            return cls()._compute_report_for_project(db_project)
+
+    @staticmethod
+    def _get_statistics_entry(statistics_object):
+        return {
+            "title": statistics_object.title,
+            "description": statistics_object.description,
+            "granularity": statistics_object.granularity,
+            "default_view": statistics_object.default_view,
+            "transformations": statistics_object.transformations,
+            "dataseries": statistics_object.calculate(),
+        }
+
+    def _compute_report_for_job(self, db_job):
+        db_report = getattr(db_job, "analytics_report", None)
+        was_created = False
+        if db_report is None:
+            db_report = AnalyticsReport.objects.create(
+                job_id=db_job.id,
+                statistics={})
+            db_report.save()
```

Comment:
This `save()` call seems to be extra, `objects.create()` should have called the DB already.